students/BrianNisonger/session4/mailroom.py

```diff
@@ -0,0 +1,91 @@
+import statistics 
+
+def sort_key(item):
+    return item[1]
+    
+def thank_you():
+    full_name="list"
+    while full_name=="list":
+        full_name=input("Please enter a Full Name==>")
+        if full_name=="list":
+            for donor in donors:
+                print (donor)
+        else:
+            if full_name not in donors:
+                donors[full_name]=[]
+            donation=input("Please enter a donation==>")
+            donation=float(donation)
+            donors[full_name].append(donation)
+            thank_you_string=f'Dear {full_name}, Thank you for your donation of {donation:.2f}.  These funds help save the migratory butterflies of New South East.  Thank you'
+            file_name=full_name.replace(" ","_")
+            file_name=f'{file_name}.txt'
+            with open(file_name,"w") as outfile:
+                outfile.write(thank_you_string)
+
+def report():
+    sum_donor={}
```

Comment:
rather than a separate dict ofr each, better to have one dict, and a structure(tuple, another dict) that gets stored for each donor.